clients/fides-js/src/lib/providers/event-context.tsx

```diff
@@ -0,0 +1,94 @@
+import { createContext, h } from "preact";
+import { MutableRefObject, ReactNode } from "preact/compat";
+import { useCallback, useContext, useMemo, useRef } from "preact/hooks";
+
+import { FidesCookie, ServingComponent } from "../consent-types";
+import {
+  dispatchFidesEvent,
+  FidesEventDetailsTrigger,
+  FidesEventExtraDetails,
+  FidesEventType,
+} from "../events";
+
+interface UseEventProps {
+  triggerRef: MutableRefObject<FidesEventDetailsTrigger>;
+  setTrigger: (newValue: FidesEventDetailsTrigger) => void;
+  servingComponentRef: MutableRefObject<ServingComponent | undefined>;
+  setServingComponent: (newValue: ServingComponent | undefined) => void;
+  dispatchFidesEventAndClearTrigger: (
+    type: FidesEventType,
+    fidesCookie: FidesCookie | undefined,
+    extraDetails?: FidesEventExtraDetails,
+  ) => void;
+}
+
+const EventContext = createContext<UseEventProps | Record<any, never>>({});
+
+export const EventProvider = ({ children }: { children: ReactNode }) => {
+  const triggerRef = useRef<FidesEventDetailsTrigger>();
```

Comment:
The provider allows us to set the Trigger info at the UI element level, and read it at the event level without needing to pass it all around through props. You'll notice this in the clean-up of the toggle components and parents.